scripts/e2etests/CreateTestDataPackage.ps1

```diff
@@ -0,0 +1,154 @@
+<#
+.SYNOPSIS
+Creates a package containing NuGet client EndToEnd test data.
+
+.PARAMETER configuration
+The build configuration.  The default value is 'debug'.
+
+.PARAMETER toolsetVersion
+The toolset version.  The default value is 15.
+
+.PARAMETER outputDirectoryPath
+The output directory where the test data package will be created.  The default value is the current directory.
+
+.EXAMPLE
+.\CreateTestDataPackage.ps1 -configuration 'debug' -toolsetVersion 15 -outputDirectoryPath 'C:\git\NuGet.Client\artifacts\nupkgs'
+#>
+
+[CmdletBinding()]
+param (
+    [ValidateSet('debug', 'release')]
+    [string] $configuration = 'debug',
+    [ValidateSet(15, 16)]
+    [int] $toolsetVersion = 15,
+    [string] $outputDirectoryPath = $PWD
+)
+
+Function Get-DirectoryPath([string[]] $pathParts)
+{
+    Return [System.IO.Path]::GetFullPath([System.IO.Path]::Combine($pathParts))
+}
+
+Set-Variable repositoryRootDirectoryPath -Option Constant -Value $(Get-DirectoryPath($PSScriptRoot, '..', '..'))
+
+Function Get-Directory([string[]] $pathParts)
+{
+    $directoryPath = Get-DirectoryPath($pathParts)
+
+    Return [System.IO.DirectoryInfo]::new($directoryPath)
+}
+
+Function Get-File([string[]] $pathParts)
+{
+    $filePath = [System.IO.Path]::Combine($pathParts)
+
+    $file = [System.IO.FileInfo]::new($filePath)
+
+    If (-Not $file.Exists)
+    {
+        throw [System.IO.FileNotFoundException]::new("Could not find $($file.Name) at $($file.FullName).  Please build first.", $file.FullName)
+    }
+
+    Return $file
+}
+
+Function Get-GenerateTestPackagesFile()
+{
+    Return Get-File($repositoryRootDirectoryPath, 'artifacts', 'GenerateTestPackages', "$toolsetVersion.0", 'bin', $configuration, 'net472', 'GenerateTestPackages.exe')
+}
+
+Function Get-NuGetFile()
+{
+    Return Get-File($repositoryRootDirectoryPath, 'artifacts', 'VS15', 'nuget.exe')
+}
+
+Function Create-TestPackages()
+{
+    $generateTestPackagesFile = Get-GenerateTestPackagesFile
+    $testPackagesDirectory = Get-Directory($repositoryRootDirectoryPath, 'test', 'EndToEnd', 'Packages')
+
+    Set-Location -Path $testPackagesDirectory.FullName
```

Comment:
Why do this?

Reply:
GenerateTestPackages.exe creates [assemblies](https://github.com/NuGet/NuGet.Client/blob/6dbccea64523830396b5690cc97da46e94fc3b26/test/TestExtensions/GenerateTestPackages/Program.cs#L251) and [packages](https://github.com/NuGet/NuGet.Client/blob/6dbccea64523830396b5690cc97da46e94fc3b26/test/TestExtensions/GenerateTestPackages/Program.cs#L81) in subdirectories of the current directory.  This statement changes the current directory to a test directory.  Currently (before this PR), the same thing is done [here](https://github.com/NuGet/NuGet.Client/blob/6dbccea64523830396b5690cc97da46e94fc3b26/test/EndToEnd/NuGet.Tests.psm1#L311).

Reply:
🤢 I wish that was configurable.

It's ok then, we shouldn't spend too much time on this anyways.

Reply:
I removed this statement because it was redundant with [this one](https://github.com/NuGet/NuGet.Client/pull/2804/files#diff-473b16bb5c7eded5d4e793380e2137beR93).